Add todo-list-common package (#10845)
backstage/backstage: docs/permission/plugin-authors/02-adding-a-basic-permission-check.md

We'll start by creating a new permission, and then we'll use the permission api to call `authorize` with it during todo creation.

## Creating a new permission

Let's navigate to the file `plugins/todo-list-common/src/permissions.ts` and add our first permission:

```diff
  import { createPermission } from '@backstage/plugin-permission-common';

- export const tempExamplePermission = createPermission({
-   name: 'temp.example.noop',
-   attributes: {},
+ export const todoListCreate = createPermission({
+   name: 'todo.list.create',
+   attributes: { action: 'create' },
  });
```

> For this tutorial, we've automatically exported all permissions from this file (see `plugins/todo-list-common/src/index.ts`). For the actual plugins that you author, we recommend exporting all permissions from your plugin, so that Backstage integrators can import them when writing policies.

Reviewer: ```suggestion
For this tutorial, we've automatically exported all permissions from this file (see `plugins/todo-list-common/src/index.ts`). For the actual plugins that you author, you must all permissions from your plugin, so that Backstage integrators can import them when writing policies.
```
